### PR TITLE
CTCP-3880: Add new EoriTCUIN form provider and update supply chain actor

### PR DESCRIPTION
**Pull request by Reece-Carruthers**

### CODE REVIEW

**Comment by Reece-Carruthers** [comment]
[screen-recorder-tue-jun-11-2024-14-58-48.webm](https://github.com/hmrc/ctc-departure-items-frontend/assets/99188015/bfe39963-5f71-4d73-b746-2db83c61cf29)

File: app/forms/mappings/Formatters.scala [hunk 1]
```diff
@@ -87,6 +88,22 @@ trait Formatters {
...
+        .flatMap {
+          eori =>
+            Right(eori.take(n).toUpperCase + eori.drop(n))
+        }
+    }
+
+    override def unbind(key: String, value: String): Map[String, String] =
+      Map(key -> value)
+  }
+
```
Comment (jackheslop96) [inline app/forms/mappings/Formatters.scala:106]: Can this be removed?

File: app/forms/mappings/Mappings.scala [hunk 2]
```diff
@@ -33,6 +33,9 @@ trait Mappings extends Formatters with Constraints {
   protected def adaptedText(errorKey: String = "error.required", args: Seq[Any] = Seq.empty)(f: String => String): FieldMapping[String] =
     of(stringFormatter(errorKey, args)(f))
 
+  protected def eoriFormat(errorKey: String = "error.required"): FieldMapping[String] =
+    of(eoriFormatter(errorKey))
```
Comment (jackheslop96) [inline app/forms/mappings/Mappings.scala:37]: Can this be removed?

**Review by jackheslop96 — APPROVED** [review]

**Comment by Reece-Carruthers** [comment]
image
image
image

**Review by jackheslop96 — APPROVED** [review]

**Review by vajarova — APPROVED** [review]

**Pull request merged**